url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/mxgraph/layout/mxCompactTreeLayout.java

```diff
@@ -0,0 +1,871 @@
+	{
+		this(graph, horizontal, false);
+	}
+
+	/**
+	 * 
+	 * @param graph
+	 * @param horizontal
+	 * @param invert
+	 */
+	public mxCompactTreeLayout(mxGraph graph, boolean horizontal, boolean invert)
+	{
+		super(graph);
+		this.horizontal = horizontal;
+		this.invert = invert;
+	}
+
+	/**
+	 * Returns a boolean indicating if the given <mxCell> should be ignored as a
+	 * vertex. This returns true if the cell has no connections.
+	 * 
+	 * @param vertex Object that represents the vertex to be tested.
+	 * @return Returns true if the vertex should be ignored.
+	 */
+	public boolean isVertexIgnored(Object vertex)
+	{
+		return super.isVertexIgnored(vertex)
+				|| graph.getConnections(vertex).length == 0;
+	}
+
+	/**
+	 * @return the horizontal
+	 */
+	public boolean isHorizontal()
+	{
+		return horizontal;
+	}
+
+	/**
+	 * @param horizontal the horizontal to set
+	 */
+	public void setHorizontal(boolean horizontal)
+	{
+		this.horizontal = horizontal;
+	}
+
+	/**
+	 * @return the invert
+	 */
+	public boolean isInvert()
+	{
+		return invert;
+	}
+
+	/**
+	 * @param invert the invert to set
+	 */
+	public void setInvert(boolean invert)
+	{
+		this.invert = invert;
+	}
+
+	/**
+	 * @return the resizeParent
+	 */
+	public boolean isResizeParent()
+	{
+		return resizeParent;
+	}
+
+	/**
+	 * @param resizeParent the resizeParent to set
+	 */
+	public void setResizeParent(boolean resizeParent)
+	{
+		this.resizeParent = resizeParent;
+	}
+
+	/**
+	 * @return the moveTree
+	 */
+	public boolean isMoveTree()
+	{
+		return moveTree;
+	}
+
+	/**
+	 * @param moveTree the moveTree to set
+	 */
+	public void setMoveTree(boolean moveTree)
+	{
+		this.moveTree = moveTree;
+	}
+
+	/**
+	 * @return the resetEdges
+	 */
+	public boolean isResetEdges()
+	{
+		return resetEdges;
+	}
+
+	/**
+	 * @param resetEdges the resetEdges to set
+	 */
+	public void setResetEdges(boolean resetEdges)
+	{
+		this.resetEdges = resetEdges;
+	}
+
+	/**
+	 * @return the levelDistance
+	 */
+	public int getLevelDistance()
+	{
+		return levelDistance;
+	}
+
+	/**
+	 * @param levelDistance the levelDistance to set
+	 */
+	public void setLevelDistance(int levelDistance)
+	{
+		this.levelDistance = levelDistance;
+	}
+
+	/**
+	 * @return the nodeDistance
+	 */
+	public int getNodeDistance()
+	{
+		return nodeDistance;
+	}
+
+	/**
+	 * @param nodeDistance the nodeDistance to set
+	 */
+	public void setNodeDistance(int nodeDistance)
+	{
+		this.nodeDistance = nodeDistance;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see com.mxgraph.layout.mxIGraphLayout#execute(java.lang.Object)
+	 */
+	public void execute(Object parent)
+	{
+		execute(parent, null);
+	}
+
+	/**
+	 * Implements <mxGraphLayout.execute>.
+	 * 
+	 * If the parent has any connected edges, then it is used as the root of
+	 * the tree. Else, <mxGraph.findTreeRoots> will be used to find a suitable
+	 * root node within the set of children of the given parent.
+	 */
+	public void execute(Object parent, Object root)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.